File: integration-tests-java/src/main/java/io/micronaut/jimmer/java/it/entity/UserRole.java
package io.micronaut.jimmer.java.it.entity;

import io.micronaut.jimmer.java.it.config.UUIDGenerator;
import io.micronaut.jimmer.java.it.config.jsonmapping.AuthUser;
import jakarta.annotation.Nullable;
import java.util.UUID;
import org.babyfish.jimmer.sql.Entity;
import org.babyfish.jimmer.sql.GeneratedValue;
import org.babyfish.jimmer.sql.Id;
import org.babyfish.jimmer.sql.LogicalDeleted;

@Entity
public interface UserRole {

    @Id
    @GeneratedValue(generatorType = UUIDGenerator.class)
    UUID id();

    String userId();

    String roleId();

    @LogicalDeleted("true")
    boolean deleteFlag();

    @Nullable
    AuthUser authUser();
}

File: integration-tests-kotlin/src/main/kotlin/io/micronaut/jimmer/kotlin/it/entity/UserRole.kt
package io.micronaut.jimmer.kotlin.it.entity

import io.micronaut.jimmer.kotlin.it.config.UUIdGenerator
import io.micronaut.jimmer.kotlin.it.config.jsonmapping.AuthUser
import org.babyfish.jimmer.sql.Entity
import org.babyfish.jimmer.sql.GeneratedValue
import org.babyfish.jimmer.sql.Id
import org.babyfish.jimmer.sql.LogicalDeleted
import java.util.UUID

@Entity
interface UserRole {
    @Id
    @GeneratedValue(generatorType = UUIdGenerator::class)
    val id: UUID

    val userId: String

    val roleId: String

    @LogicalDeleted("true")
    val deleteFlag: Boolean

    val authUser: AuthUser?
}
